addons/html_builder/static/src/plugins/navtabs_option.js

```diff
@@ -0,0 +1,118 @@
+
+class NavTabsPlugin extends Plugin {
+    static id = "NavTabsOption";
+    static dependencies = ["history", "clone", "remove"];
+    resources = {
+        builder_top_options: {
+            OptionComponent: NavTabsTopOptions,
+            selector: "section.s_tabs",
+            props: {
+                addItem: (editingElement) => this.addItem.bind(this, editingElement),
+                removeItem: (editingElement) => this.removeItem.bind(this, editingElement),
+            },
+        },
+        on_clone_handlers: this.onClone.bind(this),
+    };
+
+    get window() {
+        return this.document.defaultView;
+    }
+
+    getNavLinkEls(editingElement) {
+        return editingElement.querySelectorAll(".nav-item .nav-link");
+    }
+    getPaneEls(editingElement) {
+        return editingElement.querySelectorAll(".tab-content > .tab-pane");
+    }
+    getActiveLinkEl(editingElement) {
+        return editingElement.querySelector(".nav-link.active");
+    }
+    getActivePaneEl(editingElement) {
+        return editingElement.querySelector(".tab-pane.active");
+    }
+
+    showTab(navLinkEl, paneEl) {
+        this.window.Tab.getOrCreateInstance(navLinkEl).show();
+        // Immediately show the pane so the history remains consistent.
+        paneEl.classList.add("show");
+    }
+
+    addItem(editingElement) {
+        const activeLinkEl = this.getActiveLinkEl(editingElement);
+        const activePaneEl = this.getActivePaneEl(editingElement);
+
+        activePaneEl.classList.remove("active", "show");
+        activeLinkEl.classList.remove("active", "show");
+        const activeNavItemEl = activeLinkEl.parentElement;
+        const newNavItemEl = this.dependencies.clone.cloneElement(activeNavItemEl, {
+            historyStep: false,
+        });
+        const newPaneEl = this.dependencies.clone.cloneElement(activePaneEl, {
+            historyStep: false,
+        });
+        this.generateUniqueIDs(editingElement);
+        this.showTab(newNavItemEl.querySelector(".nav-link"), newPaneEl);
+        this.dependencies.history.addStep();
+    }
+
+    removeItem(editingElement) {
+        const activeLinkEl = this.getActiveLinkEl(editingElement);
+        const activePaneEl = this.getActivePaneEl(editingElement);
+        const nextActiveLinkEl =
+            activeLinkEl.parentElement.nextElementSibling?.firstElementChild ||
+            this.getNavLinkEls(editingElement)[0];
+        const nextActivePaneEl =
+            activePaneEl.nextElementSibling || this.getPaneEls(editingElement)[0];
+
+        this.showTab(nextActiveLinkEl, nextActivePaneEl);
+        // TODO: should this be done with the remove plugin?
+        activeLinkEl.parentElement.remove();
+        // Last to add a history step at the end.
+        this.dependencies.remove.removeElement(activePaneEl);
+    }
+
+    onClone({ cloneEl }) {
+        if (cloneEl.matches("section.s_tabs")) {
+            this.generateUniqueIDs(cloneEl);
+        }
```

Comment:
We maybe want to do it during the normalize if section.s_tabs don t have an Id, I add it ?

Reply:
What appends if cloneEl contains "section.s_tabs" ?